swift_scality_backend/diskfile.py

```diff
@@ -528,9 +526,7 @@ def reader(self, keep_cache=False):
 
         :param keep_cache:
         """
-        dr = DiskFileReader(self._filesystem, self._name,
-                            int(self._metadata['Content-Length']),
-                            self._metadata['ETag'])
+        dr = DiskFileReader(self._filesystem, self._name)
```

Comment:
No one said it still works... haha.

I relied too heavily on unit test apparently. Thanks for spotting this !